### PR TITLE
Add resources monitoring

### PR DESCRIPTION
**Pull request by Kavaldrin**
@ktf 
I made a new PR because I messed up the previous one :^(

### CODE REVIEW

**Comment by ktf** [comment]
Thanks. Looks like there is a conflict, though. Could you resolve it?

@awegrzyn do you have any news about the release?

File: Framework/Core/include/Framework/DeviceMetricsInfo.h [hunk 1]
```diff
@@ -80,6 +86,7 @@ struct DeviceMetricsInfo {
   std::vector<size_t> maxDomain;
   std::vector<MetricLabelIndex> metricLabelsIdx;
   std::vector<MetricInfo> metrics;
+  std::string deviceName;
```
Comment (ktf) [inline Framework/Core/include/Framework/DeviceMetricsInfo.h:89]: Why do you need to copy the device name?

Comment (ktf) [reply Framework/Core/include/Framework/DeviceMetricsInfo.h:89]: Ok, I think I see. Please do not copy the information, simply retrieve it from the DeviceSpec when you dump it.

**Comment by jotwinow** [comment]
Ok. @Kavaldrin Well done.

**Comment by ktf** [comment]
There is a few compilation errors, could you look into it?

**Comment by Kavaldrin** [comment]
@ktf 
It's because of version of monitoring library in alidist. New functionality requires v3.2.0 tag.

**Comment by ktf** [comment]
ok. monitoring is in. bumping this one to retrigger tests.

**Comment by jotwinow** [comment]
@ktf @Kavaldrin Giullio, it looks that a new set of tests was not triggered?

**Comment by jotwinow** [comment]
@ktf OK. Thanks. So @Kavaldrin we are done with it. Well done!

**Comment by Kavaldrin** [comment]
I hope it will be working now, Unfortunately I cannot reproduce those errors on my environment.

**Comment by ktf** [comment]
The error is:

```
Error while processing /mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Framework/Core/src/FrameworkGUIDebugger.cxx.
/mnt/mesos/sandbox/sandbox/sw/SOURCES/O2/build_o2checkcode_o2/0/Framework/Core/src/FrameworkGUIDebugger.cxx:341:34: error: format specifies type 'unsigned long long' but the argument has type 'std::array<unsigned long, 1024>::value_type' (aka 'unsigned long') [clang-diagnostic-format]
        ImGui::Text("%llu (%i)", metricsInfo.uint64Metrics[info.index][row], info.index);
                     ~~~~        ^
                     %lu
```

The safer way to format 64 bit numbers is to use `"%" PRIu64 "(%i)"`, at least until C++20.

**Comment by Kavaldrin** [comment]
@ktf Thanks for fix and for the advice :).

**Comment by jgrosseo** [comment]
@Kavaldrin I am trying to run this with the earlier syntax, i.e.:
`--resources-monitoring 1 --monitoring-backend infologger:///debug?some_prefix`
and it fails. What am I missing?
image

**Comment by ktf** [comment]
This conflicts with some change I did this morning, I think. Solution is easy, though. I will open a PR..

**Comment by ktf** [comment]
https://github.com/AliceO2Group/AliceO2/pull/3966

**Comment by Kavaldrin** [comment]
@jgrosseo You should execute task without this option '--monitoring-backend infologger:///debug?some_prefix', the default [METRIC] is required to parse metrics and then save it to JSON.

**Comment by jgrosseo** [comment]
Thanks with #3966 it got fixed.
One question though, why do the "absolute" cpu values stop after two timestamps and the others continue?
image

**Comment by Kavaldrin** [comment]
@jgrosseo  Good question, I will investigate it.

**Comment by Kavaldrin** [comment]
https://github.com/AliceO2Group/AliceO2/pull/3975 PR with fix
@ktf  There is an issue with destruction of monitoring object. The mentioned issue appeared after pulling new changes. The result is that some metrics are not sent by monitoring service, because it depends on destructor logic.

**Pull request merged**